internal/rest/dequeue_nodes.go

```diff
@@ -664,7 +668,8 @@ func (rest *RestOperations) PopulateOneCache(rest_op *utils.RestOp, aviObjKey av
 			rest.AviVsVipCacheAdd(rest_op, aviObjKey, key)
 		}
 
-	} else if rest_op.Err == nil && rest_op.Method == utils.RestDelete {
+	} else if (rest_op.Err == nil || aviErr.HttpStatusCode == 404) &&
```

Comment:
@swathinsankaran: If line number 640 set !ok, then `rest_op.Err` will be non nil and then it will hit second if condition and if `aviErr` is nil, then `aviErr.HttpStatusCode` will throw an exception. 
Should we check `ok` value also here?

Reply:
aviErr will be initialized with default values in case the type assertion fails. So the HttpStatusCode will have 0 as its value.